### DISCUSSION TITLE
Signal.h not working with ESP-IDF

### DISCUSSION THREAD

**Discussion by noName2005** [cesanta/mongoose#2427] [Ask questions, get answers] [answered]
Hello,

I'm new to programming and am currently working on an IoT project involving Mongoose for hydroponics automation. While I'm eager to integrate Mongoose, I've encountered compatibility issues with "signal.h" functions in the ESP-IDF framework. Although I've attempted refactoring to address these issues, my efforts haven't been successful.

Would you have any recommendations on how to effectively use Mongoose with ESP32 without extensive refactoring? I'm wary of potential pitfalls down the line, especially given my limited experience.

Additionally, when integrating specific portions of the Mongoose library into my project, is it sufficient to simply include "mongoose.h" at the top of my main file to access all its functionalities?

For context, my project aims to automate hydroponics. Each plant type resides in a "cell" equipped with an ESP32, sensors, and dosing pumps. While each cell can operate independently, they can also connect to a central ESP32 running an MQTT broker and a local server, enabling user control over the individual cells.

I appreciate any guidance or advice you guys have.
Warm regards, Daniel

**Comment by scaprile** [comment, marked as answer]
It is hard to understand what may probably be your problem, since we build and test our code with the ESP-IDF and it works perfectly well.
https://github.com/cesanta/mongoose/tree/master/examples/esp32/device-dashboard
https://github.com/cesanta/mongoose/tree/master/examples/esp32/uart-bridge

Please follow the guidelines in the [documentation](https://mongoose.ws/documentation), [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.

If you have an actual issue, open an issue, follow the template, and describe your issue.

PS: signal.h is not Mongoose code
